spark/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -142,12 +143,12 @@ case class OptimizeTableCommand(
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val table = getDeltaTable(child, "OPTIMIZE")
-    val txn = table.startTransaction()
```

Comment:
This is a breaking change - previously we hold a transaction before reading the snapshot, which will detect any concurrent updates ([pointer](https://github.com/delta-io/delta/blob/a24a857d27bbb4b3585814c35fa3002cee929fec/spark/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala#L181)). With this change, we could be potentially reading a stale snapshot.

I think we should guard this change behind the config you introduced. We can have a wrapper class that contains either the `txn` or the `snapshot` depending on the config and pass it down.

Reply:
nvm, I misread the code 😅 as long as later we create a txn based on this snapshot, we should be good. During commit time we try to commit with snapshot.readVersion + 1 so it should see any updates.